src/test/java/net/snowflake/client/jdbc/ResultSetAsyncIT.java

```diff
@@ -308,4 +308,25 @@ public void testIsSerializable() throws SQLException {
       // Do nothing. Test passes if we catch this exception.
     }
   }
+
+  @Test
+  public void testEmptyResultSet() throws SQLException {
```

Comment:
An empty ResultSet is returned from executeAsyncQuery, but is replaced by a real ResultSet once next() or getMetadata() is called. If neither of those functions are called, user can try to get results from empty result set but exceptions are thrown whenever a column name cannot be found.

Reply:
Add this comment to the code.